### PR TITLE
[SPARK-43849][BUILD] Enable unused imports check for Scala 2.13

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to enable `unused imports` check for Scala 2.13.

There is an exception here for `import scala.language.higherKinds`（Scala 2.13 does not need to import them , while Scala 2.12 must import them）, so this pr add two new suppression rules for it.

### Why are the changes needed?
Enable `unused imports` check for Scala 2.13.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass Github Actions

### CODE REVIEW

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -31,13 +31,9 @@ import org.apache.spark.connect.proto
 import org.apache.spark.connect.proto.{ExecutePlanResponse, SqlCommand, StreamingQueryCommand, StreamingQueryCommandResult, StreamingQueryInstanceId, WriteStreamOperationStart, WriteStreamOperationStartResult}
 import org.apache.spark.connect.proto.ExecutePlanResponse.SqlCommandResult
 import org.apache.spark.connect.proto.Parse.ParseFormat
-import org.apache.spark.connect.proto.StreamingQueryCommand
```
Comment (LuciferYang) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:34]: Scala 2.13 checked them unused as they have already been imports in line 31

File: sql/catalyst/src/main/scala-2.13/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 package org.apache.spark.sql.catalyst.expressions
 
-import scala.collection.{mutable, IterableFactory, IterableOps}
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala-2.13/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala:20]: `IterableFactory` and `IterableOps` are unused imports

**Comment by LuciferYang** [comment]
cc @dongjoon-hyun  FYI

File: pom.xml [hunk 3]
```diff
@@ -2891,6 +2891,13 @@
               <arg>-Wconf:cat=unchecked&amp;msg=outer reference:s</arg>
               <arg>-Wconf:cat=unchecked&amp;msg=eliminated by erasure:s</arg>
               <arg>-Wconf:msg=^(?=.*?a value of type)(?=.*?cannot also be).+$:s</arg>
+
```
Comment (dongjoon-hyun) [inline pom.xml:2894]: nit. Redundant empty line.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs)

File: pom.xml (hunk 3 above)
Comment (LuciferYang) [reply pom.xml]: Thanks @dongjoon-hyun , [e80f584](https://github.com/apache/spark/pull/41356/commits/e80f584da7ee5beff27ca9209e81488aa285c3c0) removed it

**Review by MaxGekk — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun @MaxGekk

**Pull request closed without merging**